src/java/org/apache/nutch/crawl/CrawlDbReader.java

```diff
@@ -185,13 +190,83 @@ public synchronized void write(Text key, CrawlDatum value)
         out.writeByte('\n');
       }
 
-      public synchronized void close(TaskAttemptContext context) throws IOException {
+      public synchronized void close(TaskAttemptContext context)
+          throws IOException {
+        out.close();
+      }
+    }
+
+    public RecordWriter<Text, CrawlDatum> getRecordWriter(
+        TaskAttemptContext context) throws IOException {
+      String name = getUniqueFile(context, "part", "");
+      Path dir = FileOutputFormat.getOutputPath(context);
+      FileSystem fs = dir.getFileSystem(context.getConfiguration());
+      DataOutputStream fileOut = fs.create(new Path(dir, name), context);
+      return new LineRecordWriter(fileOut);
+    }
+  }
+
+  public static class CrawlDatumJsonOutputFormat
+      extends FileOutputFormat<Text, CrawlDatum> {
+    protected static class LineRecordWriter
+        extends RecordWriter<Text, CrawlDatum> {
+      private DataOutputStream out;
+      private ArrayList<String> jsonString = new ArrayList<String>();
+      public LineRecordWriter(DataOutputStream out) {
+        this.out = out;
+        try {
+          out.writeBytes("[");
+        } catch (IOException e) {
+        }
+      }
+
+      public synchronized void write(Text key, CrawlDatum value)
+          throws IOException {
+        String fetchTime = new Date(value.getFetchTime()).toString();
+        String modifiedTime = new Date(value.getModifiedTime()).toString();
+        String recordString = "";
+        recordString += "{\n" + "\t\"url\":\"" + key.toString()
```

Comment:
I would strongly recommend to use a JSON library to do the escaping. There are many pitfalls, e.g., what if a URL contains a `"`? [Jackson](https://github.com/FasterXML/jackson) is already on board (it's a dependency) and it's pretty handy to write JSON:
```java
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectWriter;
...
public static class JsonIndenter extends MinimalPrettyPrinter {

    // @Override
    public void writeObjectFieldValueSeparator(JsonGenerator jg)
        throws IOException, JsonGenerationException {
      jg.writeRaw(": ");
    }

    // @Override
    public void writeObjectEntrySeparator(JsonGenerator jg)
        throws IOException, JsonGenerationException {
      jg.writeRaw(", ");
    }
}
...
// inside write(Text key, CrawlDatum value)
ObjectMapper jsonMapper = new ObjectMapper();
jsonMapper.getFactory().configure(JsonGenerator.Feature.ESCAPE_NON_ASCII,
        true);
ObjectWriter jsonWriter = jsonMapper.writer(new JsonIndenter());

...
Map<String, String> data = new LinkedHashMap<String, String>(); // a linked hash map preserves the ordering
data.put("url", key.toString());
// put all other fields into the map
...
// and write the serialized record
out.write(jsonWriter.writeValueAsBytes(data));
```

There are many options to control the output format...